tensorflow/compiler/xla/service/gpu/launch_dimensions.cc

```diff
@@ -116,7 +133,12 @@ StatusOr<LaunchDimensions> CalculateLaunchDimensions(
       "block) = ceil(%d/%d) = %d",
       num_elements, threads_per_block, block_count);
 
-  return LaunchDimensions(block_count, threads_per_block);
+  bool last_dim_aligned_for_vectorization = false;
```

Comment:
That seems to be a layering violation, LaunchDimensions is meant to encapsulate how are we launching the kernel, it normally does not affect the codegen?

Reply:
There is inter-dependency here. I removed that and added a long condition in parallel_loop_emitter.cc instead.